core/src/avm2/regexp.rs

```diff
@@ -204,17 +209,68 @@ impl<'gc> RegExp<'gc> {
         ret
     }
 
-    pub fn replace(
+    // Implements string.replace(regex, replacement) where the replacement is
+    // a function.
+    pub fn replace_fn(
+        &mut self,
+        activation: &mut Activation<'_, 'gc, '_>,
+        text: AvmString<'gc>,
+        f: &FunctionObject<'gc>,
+    ) -> AvmString<'gc> {
+        self.replace_with_fn(activation, &text, |activation, txt, m| {
+            let mut args = std::iter::once(Some(&m.range))
+                .chain((m.captures.iter()).map(|x| x.as_ref()))
+                .map(|o| match o {
+                    Some(r) => Value::from(AvmString::new(
+                        activation.context.gc_context,
+                        &txt[r.start..r.end],
+                    )),
+                    None => Value::from(AvmString::default()),
+                })
+                .collect::<Vec<_>>();
+            args.push(Value::from(m.range.start));
+            args.push(Value::from(*txt));
+
+            let r = f.call(activation.global_scope(), &args, activation);
```

Comment:
I'm afraid we'd want to propagate exception here. That means the callbacks passed to `replace_with_fn` will need return `Result<WString, Error>`.

Reply:
Done. I did not find a way to create an error so that this could be tested though.